uk.co.compucorp.civicrm.hrcore/CRM/HRCore/SearchTask/SearchTaskAdderInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+interface CRM_HRCore_SearchTask_SearchTaskAdderInterface {
+  /**
+   * Add items to the tasks array in the form:
+   * [
+   *   'title' => <the title to be displayed in the dropdown>,
+   *   'class' => <the class name of the form for the action>
+   * ]
+   *
+   * @param $tasks
+   *
+   * @return void
+   */
+  public static function add(&$tasks);
+
+  /**
+   * Determines whether or not the tasks should be added.
+   *
+   * @param $objectName
+   *
+   * @return bool
+   */
+  public static function shouldAdd($objectName);
+}
```

Comment:
I don't see the point of it either for this case, but I got the impression that you wanted this to be very future proof for other developers coming along. If the interface is there the next dev is more likely to copy it, and in that case we'll have a consistent approach.